File: cpp/11.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i=0; 
        int j=height.size()-1;
        int mx = 0;
        while(i!=j) {
            if(height[i]<height[j]) {
                mx = max(mx, (j-i)*height[i]);
                i++;
            }
            else {
                mx = max(mx, (j-i)*height[j]);
                j--;
            }
        }
        return mx;

    }
};

File: python/11.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i=0
        j=len(height)-1
        mx = 0

        while(i<j):
            if(height[i]<height[j]):
                mx = max(mx, height[i]*(j-i))
                i+=1
            else:
                mx = max(mx, height[j]*(j-i))
                j-=1

        return mx